projects/stargazer/plugins/other/radius/server.cpp

```diff
@@ -54,22 +56,28 @@ RadProto::Packet Server::makeResponse(const RadProto::Packet& request)
 void Server::handleSend(const error_code& ec)
 {
     if (ec)
-        std::cout << "Error asyncSend: " << ec.message() << "\n";
+    {
+        m_logger("Error asyncSend: %s", ec.message());
+        printfd(__FILE__, "Error asyncSend: '%s'\n", ec.message());
+    }
 
-    startReceive();
+    if (!m_token.stop_requested())
```

Comment:
I think you should check the token before checking the error code. When you close the socket and there is a pending send or receive operation, it will result in error. However, when we are stopping, this error is expected and is not something we want to report in the log.